src/channel/bounded.rs

```diff
@@ -79,6 +85,8 @@ mod inner {
     pub struct Sender<T>(pub(super) InnerSender<T>);
     /// A bounded receiver, created with [`channel`]
     pub struct Receiver<T>(pub(super) InnerReceiver<T>);
+    /// A bounded stream, created with a channel
+    pub struct BoundedStream<T>(pub(super) InnerReceiver<T>);
```

Comment:
Shouldn't this be a stream type, like the `RecvStream` in the flume channel, rather than an `InnerReceiver`?

Reply:
Hmm looks like we also use the receiver directly in `fn into_stream` for the async std channel, but convert the receiver to a stream for other channel types, so I guess the async std channel is indeed different. Not sure why though.

Reply:
Not sure why, but async-std chooses to [implement `Stream` directly on the `Receiver` type](https://docs.rs/async-std/latest/async_std/channel/struct.Receiver.html). So, I treated the `Receiver` as the stream. I'm also not really sure why flume requires a lifetime, but none of the other channels do.. Maybe @nyospe has some intuition?